### PR TITLE
Standardize habitable ranges

### PR DESCRIPTION
**Pull request by Amazinite**
**Content (Artwork / Missions / Jobs)**

## Summary
This PR standardizes habitable ranges across all systems given the star(s) that are in that system, resolving #6359.
The following habitable ranges have been used:

Star type | Habitable range | Notes
---|---|---
o-supergiant | 81920 | 2x o-giant.
o-giant | 40960 | Proportional to image size of o0.
o0 | 20480 | Radius of 80.
o3 | 17560 | Radius of ~76.
o5 | 15560 | Radius of ~73.
o8 | 13720 | Radius of 70.
o-dwarf | 2100 | Proportional to image size of o8.
b-supergiant | 46000 | 2x b-giant.
b-giant | 23000 | Proportional to image size of b0.
b0 | 11500 | Radius of 66.
b3 | 10000 | Radius of ~63.
b5 | 8640 | Preexisting value. Radius of 60.
b8 | 6600 | Radius of ~55.
b-dwarf | 1485 | Proportional to image size of b8.
a-supergiant | 24000 | 2x a-giant.
a-giant | 12000 | Proportional to image size of a0.
a0 | 5000 | Preexisting value. Radius of 50.
a3 | 4200 | Radius of ~47.
a5 | 3645 | Preexisting value. Radius of 45.
a8 | 3000 | Radius of ~42.
a-dwarf | 750 | Proportional to image size of a8.
f-supergiant | 13900 | 2x f-giant.
f-giant | 6950 | Proportional to image size of f0.
f0 | 2560 | Preexisting value. Radius of 40.
f3 | 2200 | Radius of 38.
f5 | 1715 | Preexisting value. Radius of 35.
f5-old | 3430 | #5828 increased the size of "-old" stars compared to their normal variant; 2x f5.
f8 | 1310 | Radius of 32.
f-dwarf | 355 | Proportional to image size of f8.
g-supergiant | 8640 | 2x g-giant.
g-giant | 4320 | Proportional to image size of g0.
g0 | 1080 | Preexsiting value. Radius of 30.
g0-old | 2160 | #5828 increased the size of "-old" stars compared to their normal variant; 2x g0.
g3 | 700 | g5 stars in ringworld systems had this value. All ringworld systems now use g3 stars. Radius of 26.
g5 | 625 | Preexisting value. Radius of 25.
g5-old | 1250 | #5828 increased the size of "-old" stars compared to their normal variant; 2x g5.
g8 | 550 | Radius of 24.
g-dwarf | 150 | Proportional to image size of g8.
k-supergiant | 6000 | 2x k-giant.
k-giant | 3000 | Proportional to image size of k0.
k0 | 490 | Radius of ~23.
k0-old | 980 | #5828 increased the size of "-old" stars compared to their normal variant; 2x k0.
k3 | 450 | Radius of ~22.5
k5 | 425 | Was sometimes 320, sometimes ~425. Chose 425. Radius of ~22.
k5-old | 950 | #5828 increased the size of "-old" stars compared to their normal variant; 2x k5.
k8 | 370 | Radius of ~21.
k-dwarf | 100 | Proportional to image size of k8.
m-supergiant | 4600 | 2x m-giant.
m-giant | 2300 | Proportional to image size of m0.
m0 | 320 | Was sometimes 320, sometimes ~220. Chose 320; radius of 20.
m3 | 230 | Radius of ~18.
m5 | 160 | Radius of ~16.
m8 | 135 | Was sometimes 135, sometimes ~80. Chose 135; radius of 15.
m-dwarf | 35 | Proportional to image size of m8.
l-dwarf | 30 | Just a value less than m-dwarves.
rouge brown dwarves | 10 | Just a tiny value less than proper stars.
wr | 625 | Preexisting value.
carbon | 625 | Matching the wr value; subject to future change.
nova | 100 | Preexisting value.
nova-old | 100 | Preexisting value.
neutron | 100 | Neutron stars are unused; matching the nova value.
black-hole-core | 10000 | Preexisting value.

As noted, some stars had preexisting values that were already used. Where a preexisting value existed, I have used it. Where two preexisting values had existed, I chose one that had the impact on the least number of systems. (For example m8 stars sometimes had a habitable range of 135 and at other times it was about 80, but it was more often 135 that was used.) 

The way the system editor ties habitable range to a star is by defining a "radius" for each star. The habitable range for a star is then `(radius^3) / 25`. The "radius" of a star before was quite literally the radius of the star in its image, but #5828 changed the radius on the image of many of the preexisting stars, so instead of remeasuring new radii and potentially having to change more systems, I stuck with the radii that were used before. Then as I filled in stars between preexisting ones I just chose radius values between the values I already had, and for larger stars I just increased from the previous highest value. As can be seen, as the stars get bigger, the radius used gets bigger faster, hence the extremely large habitable radii on many of the larger stars. Given that the actual image radius is now divorced from the habitable range, when the system editor is updated the habitable range of each star should be a given constant instead of a value derived from the radius.

This "radius" system for defining a star's habitable range was only used on the basic stars, though. (i.e. the stars named some letter + (0 | 3 | 5 | 8). Variants of these stars used a different method: 
* For the giant and dwarf variants, each have habitable ranges proportional with the width/height of the 0 and 8 variant images respectively. That is, if a giant variant is 100x100 and its corresponding 0 variant is 50x50, then the giant is 2x the habitable range of the 0 variant. I chose to do this instead of going by an area (which would have been 4x) because doing so would have resulted in basically unusable habitable ranges for most if not all giants. 
* Supergiant variants are simply double the habitable range of the giant variant of the same type.
* For the old variants, since #5828 increased the size of the old variants compared to the normal variant of the same type (e.g. g0 vs g0-old), I've made the old variant habitable ranges double that of the normal variant. Previously the normal and old variants had the same habitable ranges, so this greatly increased the habitable range on any system with old stars.

After deciding on the habitable ranges for each star type, I updated my formatter to change the habitable range of every system to match its stars, and then output the change in habitable range for each system. For example, the system 1 Axis outputted as `1 Axis	745.92	745` where the first value is the old range and the second value is the new one. I then pasted the results of all systems into a spreadsheet and ordered by the % change made to each system. I then manually checked all systems that had a % change in their habitable range >5%.

For most of these systems I was able to simply change the star types used to more suitable stars, for example by changing all g5-old stars in ringworld systems to g3 stars, drastically bringing down the % change made to those systems, in some cases even bringing it down to 0%. For the remaining systems, if the system was uninhabited then I didn't bother with changing the star(s), but may have changed the sprites used on some of the planets in the system, such as avoiding using ice planets near very hot stars. If the system was inhabited then I either moved the inhabited planet in the system, changed the star(s) used, or both. 

When I started looking at systems with >100% changes, I ran into a few problem systems that I'll highlight here. Many of the >100% change systems were uninhabited, but some were inhabited systems that had presumably been changed to better match the star types of the IRL system of the same name.
* Polaris wasn't really a problem system, but has been changed heavily to better match the IRL star.
* Almaaz was changed from using an f-supergiant to an f-giant, and Freedom was moved much farther out than it was to be at a more habitable distance. The f-supergiant habitable range I came up with is beyond the system fence and therefore unfit for use in an inhabited system.
* Prime in Betelgeuse was moved to a more habitable distance. The star is still a red supergiant.
* Featherweight in Mebsuta was moved to a more habitable distance. The stars are unchanged.
* Suhail was changed from a k-supergiant to a k-giant. The planet in the system didn't need any other changes; it's a little on the hot side, but it already was before and is highlighted in its description as not particularly habitable.
* Alpha Centauri in #5828 was made a k0/g3 pair with the g3 star orbiting much further out. Since habitable ranges are only ever from system center, I've pushed the stars together, and then moved Chiron to a more habitable distance.
* Changed Algol from an a0/a5 pair to a a8/k0 pair. Not sure what the reasoning behind the original pair was, but the great Wikipedia page for Algol tells me that it's a trinary system consisting of a b8, k0, and a7 star. The first two are actually the closer ones, but I've decided to go with the cooler two for the sake of not needing a massive habitable range. Quicksilver has been moved accordingly.
* Downgraded Naos from an o5 to an f8; this is a mismatch from the real world star of the same name, but the habitable range on an o5 is far beyond the system fence and therefore basically unusable for an inhabited system like Naos.
* Downgraded Alnitak from an o-supergiant/b0 pair to an f8 star for the same reasons; the habitable range of the original pair was more than 11x farther than the system fence. This is again a mismatch from the IRL star system.

Finally, I updated the formatter to adjust the orbital periods of all objects in each system. This included adjusting the orbital periods of binary star systems, which also required adjusting the distances; I've manually checked all binary star systems to ensure that the new distances don't cause the stars to clip one another or to clip nearby planets.

The system editor uses the same radius used to determine a star's habitable range to determine its mass, which is in turn what determines the orbital periods of objects in the system. The issue with this though is that while habitable range is a measure of an object's temperature, not all objects have a temperature that matches their mass. For example, nova remnants and neutron stars have habitable ranges of 100, but they certainly have masses higher than an equivalent star with a habitable range of 100. So the neutron star and nova remnants now use a mass equivalent to an a0 star, the black hole uses a mass double that, and the rouge brown dwarves use a mass double their habitable range.

Compared to before this PR, here is a summary of the changes made in terms of the % change in habitable range of systems. Out of 402 systems:
* 152 receive no change.
* 231 receive a <=1% change.
* 293 receive a <=5% change; above this is where manual changes have likely been made to planet distances.
* 318 receive a <=10% change.
* 339 receive a <=25% change.
* 347 receive a <=50% change.
* 382 receive a <=100% change; above this is where significant changes may have been made.
* 020 receive a >100% change.

I'll note that some of the >100% changes were also because the star used simply terribly mismatched the habitable range that was used, for example Queri using a range of 135 on a g5 star.

## Save File
N/A

## PR Checklist
N/A

### CODE REVIEW

**Comment by MasterOfGrey** [comment]
Hot damn, this is a lot. This is excellent though.

**Comment by Amazinite** [comment]
Marking as a draft because there are still a few things I want to clean up:
- [ ] Rework the systems that use a brown dwarf as the system center instead of a star. Such systems should have less objects than a typical system, and the objects they do have should orbit close to the center. Since brown dwarves are closer to gas giants than stars, perhaps they don't have gas giants orbiting them.
- [ ] The habitable range chosen for carbon stars was pulled out of no where. Decide on a better habitable range for carbon stars.
- [ ] Although neutron stars are unused at the moment, we could likely decide on a better habitable range for them as well for when they do get used.

**Comment by MasterOfGrey** [comment]
> * [ ]  Although neutron stars are unused at the moment, we could likely decide on a better habitable range for them as well for when they do get used.

Neither brown dwarves nor neutron stars should effectively have habitable ranges.

Brown dwarves are lucky if they emit enough in the infrared to hold off freezing atmosphere on rocky-sized worlds, with minimal solar output, and neutron stars emit enough violently ionising radiation during their formation and pretty much any post-formation interaction with normal matter that they'd thoroughly sterilise anything close enough to be warmed by them.

Nothing even vaguely resembling life compatible with the human experience would be found on a world surrounding either of them.

**Comment by Amazinite** [comment]
The "habitable" distance is really a measurement of an object's temperature. While yes, the ionizing radiation from a neutron star would make it impossible for anything to inhabit a planet around it, that wouldn't affect its habitable distance, which is based off of its temperature.

Since we're already using 100 for novas, I'm just going to use 100 for neutron stars as well.
And as far as brown dwarves go, I already have those with a habitable range of 10, far smaller than anything other object.

**Comment by Amazinite** [comment]
The last star-related thing I had in mind but that I think I'll save for a separate PR: I didn't get around to reviewing #5828 before it was merged, but aside from the habitable distances issue that I wrote down after that PR was merged that this PR addresses, the only other issue I had with that PR was that it changed the WR star image. 

I have two problems with the new image:
1. It's far too bright to me.
2. The WR star now mismatches the color of the star in the Sad Archie scene image.

<details><summary>Images for comparison:</summary>

Old:
image

New:
wr+

Sad Archie scene:
image

</details>

Instead of just outright removing the new WR image we could have a wr0 and wr1 and use the older image for Ildaria, but if others agree that the newer image is way too bright then we could just revert back to the old one entirely.

**Comment by Zitchas** [comment]
I personally like both old and new image, so I'd be a fan of using the old one for Ildaria and leaving the new one for other places.

That being said, I do wonder if there's anything else at play. I'm generally pretty sensitive to brightness/contrast, and I don't have any issues with it.

**Comment by MasterOfGrey** [comment]
I believe I was in support of both of them being used in places as needed on the original PR as well.

File: data/map.txt [hunk 1]
```diff
@@ -18785,29 +18710,21 @@ system Perfica
...
 		distance 962
-		period 348
-	object
-		sprite planet/desert4-b
-		distance 1757.75
-		period 896.982
-	object
-		sprite planet/gas4
-		distance 2712.71
-		period 1719.7
```
Comment (Zitchas) [inline data/map.txt]: Is there a reason why these two objects were removed here?

Comment (Amazinite) [reply data/map.txt]: This is a brown dwarf system, and I reduced the number of objects in the few systems that had a brown dwarf instead of a star.

Comment (Zitchas) [reply data/map.txt]: Alright. Thanks for the explanation.

**Pull request merged**